### ISSUE TITLE
[UpdateNotify] Cog sends notification about newer version for older versions

### ISSUE THREAD

**Issue by jack1142** [PhasecoreX/PCXCogs#23]
While I don't use that cog, I noticed that it may notify about "update" when the version installed is newer than the one that pypi reports, e.g.
```
Hello!

There is a newer version of Red-DiscordBot available!
Your version: 3.4.0
Latest version: 3.3.10
```

**Comment by PhasecoreX** [comment]
When you got this notification, was the latest version on PyPI 3.3.10 at the time? UpdateNotify simply compares the PyPI version string to the currently installed Redbot version string. If you were to install a newer version (or maybe a dev version?) before it made it to PyPI, it would probably look like this.

Or was 3.4.0 out on PyPI and for some reason the cog reported 3.3.10 as the latest, even though it came out a month ago? As well as 3.3.11 and 3.3.12 coming out after it as well. If that's the case... that's really weird...

I'm just using this: https://pypi.org/pypi/Red-DiscordBot/json, looking at `info.version` It should be the latest always.

**Comment by jack1142** [comment]
As I said in the issue description, I don't use the cog; this issue appeared in Red's main server in #support. As far as I can tell from the data I gathered there though, the message was sent about 16 hours ago, which was a day *after* 3.4 has been released and the user has already updated to 3.4.0. It's not dev version either as version numbers of those look like this: `3.4.0.dev1`.

**Comment by jack1142** [comment]
Sorry, I misclicked; I edited the comment I posted to contain everything I wanted to say 😄

**Comment by PhasecoreX** [comment]
Yes, I am going to look into a more robust version checking method. Thank you for pointing out `VersionInfo`!
